### PR TITLE
Move test suites to new test-suites-base project

### LINKED ISSUE #26746: Pull out test suites to its own project
In preparation for multi-target test suites, the current classes should all be written in Java 8. However, `testing-base` is Java 6 for workers. We should be able to pull out the test suite classes to a new project to rectify this.

### PR DESCRIPTION
**Pull request by octylFractal**
Fixes #26746

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

File: platforms/software/test-suites-base/src/main/java/org/gradle/testing/base/plugins/package-info.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Plugin classes for generic support for testing.
+ */
+package org.gradle.testing.base.plugins;
```
Comment (tresat) [inline platforms/software/test-suites-base/src/main/java/org/gradle/testing/base/plugins/package-info.java:20]: ❓ Shouldn't this be failing without an `org.gradle.api.NonNullApi` annotation per Alex's recent change?

**Review by tresat — APPROVED** [review]

File: platforms/software/test-suites-base/src/main/java/org/gradle/testing/base/plugins/package-info.java (hunk 1 above)
Comment (octylFractal) [reply platforms/software/test-suites-base/src/main/java/org/gradle/testing/base/plugins/package-info.java:20]: Do you mean the unmerged https://github.com/gradle/gradle/pull/26593 ?

Comment (tresat) [reply platforms/software/test-suites-base/src/main/java/org/gradle/testing/base/plugins/package-info.java:20]: Yes.  My mistake, I assumed it was merged by now.  I think it's still worth annotating the package in advance.

**Comment by octylFractal** [comment]
@bot-gradle test and merge

File: platforms/software/test-suites-base/src/main/java/org/gradle/testing/base/plugins/package-info.java (hunk 1 above)
Comment (octylFractal) [reply platforms/software/test-suites-base/src/main/java/org/gradle/testing/base/plugins/package-info.java:21]: I did so.

**Comment by octylFractal** [comment]
@bot-gradle test and merge

**Pull request merged**